### PR TITLE
Fixed mrtk_development compiler & build errors

### PR DESCRIPTION
**Pull request by StephenHodgson**
Overview
---
Fixed some compiler errors, and cleaned up collection classes a bit to conform to established coding guidelines and patterns.

Changes
---
- Fixes: #2776

### CODE REVIEW

File: Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseObjectCollection.cs [hunk 1]
```diff
@@ -1,41 +1,28 @@
...
-        public List<ObjectCollectionNode> NodeList
-        {
-            get { return nodeList; }
-            set { nodeList = value; }
-        }
+        protected List<ObjectCollectionNode> NodeList { get; } = new List<ObjectCollectionNode>();
 
         [Tooltip("Whether to include space for inactive transforms in the layout")]
         [SerializeField]
-        protected bool ignoreInactiveTransforms = true;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseObjectCollection.cs]: Why are you changing the visibility of these? Classes may need to access them.

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseObjectCollection.cs]: I'm not. The property accessor is public.

Comment (StephenHodgson) [reply Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Collections/BaseObjectCollection.cs]: Besides it's always easier to give access than to take it away.

**Comment by StephenHodgson** [comment]
Cool, let's get gestures approved too! ;p

**Pull request merged**